### DISCUSSION TITLE
How to cancel sorting of DataGrid by click the column?

### DISCUSSION THREAD

**Discussion by GF-Huang** [HandyOrg/HandyControl#722] [Q&A]
When I click the column for sorting, how can I cancel it? In other words, back to unsorted state by UI.

**Comment by ghost1372** [comment]
mybe this can help you
`CollectionViewSource.GetDefaultView(datagrid.ItemsSource).SortDescriptions.Clear();
`

**Reply by GF-Huang** [reply]
But how to make it in a state loop? Unsorted -> Ascending -> Descending -> Unsorted...

**Reply by ghost1372** [reply]
mybe Disable sort by `CanUserSort="False"` and implement new logic You may also need to change the styles
